### ISSUE TITLE
Is it correct mask for batch generation in implementation trl.Trainer.generate?

### ISSUE THREAD

**Issue by knowledgehacker** [lvwerra/trl#344]
I read trl.Trainer.generate and find that if the input is a list of tensors, it call _generate_batch, which creates attention_mask for the input_ids, it just mask every id as a valid one, none is a padding one, I don't think it is the correct mask.
If the input is a single tensor, it calls underlying pre-trained model's generate method, it will create correct attention_mask for the input_ids with the pad_token_id passed in generation_kwargs.

Would someone take a look at it? Thanks

**Comment by younesbelkada** [comment]
Hi @knowledgehacker 
Indeed we do create attention masks as all ones inside `_generate_batched`. However this seems to be needed to create the correct attention mask by the `tokenizer.pad` method. 
If you retrieve the attention mask from `padded_inputs["attention_mask"]`, you'll be able to see the correct attention masks
```bash
padded_inputs["attention_mask"]
tensor([[1, 1, 1, 1, 1, 1, 0],
        [1, 1, 1, 1, 1, 0, 0],
        [1, 1, 1, 1, 1, 1, 1],
        [1, 1, 1, 1, 1, 1, 0]], device='cuda:0')
```
